src/data/repositories/Dhis2ConfigRepository.ts

```diff
@@ -46,55 +49,61 @@ export class Dhis2ConfigRepository implements ConfigRepository {
-        if (version > 30 && d2User.settings.keyUiLocale) {
-            return d2User.settings.keyUiLocale;
-        }
+    public getInstance(): Instance {
+        return this.instance;
+    }
 
-        const settings = await this.api.get<{ keyUiLocale: string }>("/userSettings").getData();
-        return settings.keyUiLocale ?? "en";
+    private async getConfig() {
+        return this.storageClient.getObject<PersistedConfig>(Namespaces.CONFIG);
     }
 
-    public getInstance(): Instance {
-        return this.instance;
+    public async get(): Promise<PersistedConfig> {
+        const config = (await this.getConfig()) ?? {};
+        return {
+            ...defaultConfig,
+            ...config,
+        };
     }
 
-    public async getSettingsPermissions(): Promise<Permission> {
+    public async save(update: PersistedConfig): Promise<PersistedConfig> {
         const config = await this.getConfig();
-        const { users = [], userGroups = [] } = config.settingsPermissions ?? {};
-        return { users, userGroups };
+        const updatedConfig = {
+            ...config,
+            ...update,
+        };
+
+        return this.storageClient
+            .saveObject<PersistedConfig>(Namespaces.CONFIG, updatedConfig)
+            .then(() => updatedConfig);
     }
 
-    public async updateSettingsPermissions(update: Partial<Permission>): Promise<void> {
+    public async importTranslations(language: string, terms: Record<string, string>): Promise<TranslatableText[]> {
         const config = await this.getConfig();
-        const { users = [], userGroups = [] } = config.settingsPermissions ?? {};
 
-        await this.storageClient.saveObject<PersistedConfig>(Namespaces.CONFIG, {
-            ...config,
-            settingsPermissions: {
-                users: update.users ?? users,
-                userGroups: update.userGroups ?? userGroups,
-            },
-        });
-    }
+        if (!config?.customText) return [];
+        const { customText } = config;
 
-    public async getShowAllModules(): Promise<boolean> {
-        const { showAllModules = true } = await this.getConfig();
-        return showAllModules;
+        const translatedText: Partial<CustomText> = {
+            ...customText,
+            root_title: customText.root_title
+                ? importTranslate(customText.root_title, language, terms[customText.root_title.key])
+                : undefined,
+            root_subtitle: customText.root_subtitle
+                ? importTranslate(customText.root_subtitle, language, terms[customText.root_subtitle.key])
+                : undefined,
+        };
+
+        const updatedConfig = await this.save({ customText: translatedText });
+
+        return this.extractTranslatableText(updatedConfig);
     }
 
-    public async setShowAllModules(showAllModules: boolean): Promise<void> {
+    public async extractTranslations(): Promise<TranslatableText[]> {
         const config = await this.getConfig();
```

Comment:
Not sure about this but, could we use the `get()` instead of the `getConfig()` so we don't need to check if the config exists? or is it an issue having the default config? (also this would also apply on other spots where we are using the `getConfig()`)